include/foonathan/memory/allocator_storage.hpp

```diff
@@ -126,7 +126,7 @@ namespace foonathan
                 FOONATHAN_REQUIRES(
                     (!std::is_base_of<allocator_storage, typename std::decay<Alloc>::type>::value))>
             allocator_storage(Alloc&& alloc,
-                              FOONATHAN_SFINAE(new storage_policy(detail::forward<Alloc>(alloc))))
+                              FOONATHAN_SFINAE(new storage_policy(std::declval<Alloc&&>())))
```

Comment:
Nitpick: the && is not necessary, `std::declval` adds it already: https://en.cppreference.com/w/cpp/utility/declval

Reply:
Thx, I didn't know! I always thought that `std::declval` would return a value unless you gave it a reference type.